digitalocean/config/config.go

```diff
@@ -76,30 +75,37 @@ func (c *Config) Client() (*CombinedConfig, error) {
 
 	userAgent := fmt.Sprintf("Terraform/%s", c.TerraformVersion)
 	var client *http.Client
+	var godoOpts []godo.ClientOpt
+
+	client = oauth2.NewClient(context.Background(), tokenSrc)
 
 	if c.HTTPRetryMax > 0 {
-		retryableClient := retryablehttp.NewClient()
-		retryableClient.RetryMax = c.HTTPRetryMax
-		retryableClient.RetryWaitMin = time.Duration(c.HTTPRetryWaitMin * float64(time.Second))
-		retryableClient.RetryWaitMax = time.Duration(c.HTTPRetryWaitMax * float64(time.Second))
+		retryConfig := godo.RetryConfig{
+			RetryMax:     c.HTTPRetryMax,
+			RetryWaitMin: godo.PtrTo(c.HTTPRetryWaitMin),
+			RetryWaitMax: godo.PtrTo(c.HTTPRetryWaitMax),
+			Logger:       log.New(os.Stderr, "", log.LstdFlags),
```

Comment:
```suggestion
			Logger:      log.Default(),
```

With some additional testing, it looks like `log.Default()` will work a bit better in this case. With my original suggestion, it was still making it to the debug logs, but for some reason it was wrapped in an "unexpected data" warning.

```
2023-08-17T15:51:37.311-0400 [WARN]  unexpected data: registry.terraform.io/digitalocean/digitalocean:stderr="2023/08/17 15:51:37 [DEBUG] GET https://api.digitalocean.com/v2/domains/example-c4a4a3d0-3d1d-11ee-9746-43ce1079c844.com/records/1694880375"
```